crates/sound/src/audio_backends/pw_backend.rs

```diff
@@ -133,16 +160,27 @@ impl PwBackend {
         println!("pipewire backend running");
 
         let streams_param = vec![PCMParams::default(); NR_STREAMS];
+        let streams_states = vec![0; NR_STREAMS];
 
         Self {
             thread_loop,
             core,
+            context,
             stream_params: RwLock::new(streams_param),
+            stream_states: RwLock::new(streams_states),
+            stream_hash: RwLock::new(HashMap::with_capacity(NR_STREAMS)),
+            stream_listener: RwLock::new(HashMap::with_capacity(NR_STREAMS)),
         }
     }
 }
 
 impl AudioBackend for PwBackend {
+    fn get_stream_state(&self, stream_id: u32) -> Result<u32> {
+        let stream_states = self.stream_states.read().unwrap();
+        dbg!("stream states: {}", stream_states.clone());
+        Ok(stream_states[stream_id as usize])
```

Comment:
If a malicious guest sends a wrong stream_id, could this generate an out-of-boundary panic?

Reply:
maybe it might make sense to propagate the error here